### PR TITLE
Implement Append for Function1

### PR DESCRIPTION
**Pull request by dwijnand**
Supersedes #4519

### CODE REVIEW

File: main-settings/src/main/scala/sbt/Append.scala [hunk 1]
```diff
@@ -8,110 +8,106 @@
...
+      def appendValues(a: Seq[T], b: Seq[V]): Seq[T] = a ++ b.map(x => (x: T))
       def appendValue(a: Seq[T], b: V): Seq[T] = a :+ (b: T)
     }
+
   @compileTimeOnly("This can be used in += only.")
   implicit def appendTaskValueSeq[T, V <: T]: Value[Seq[Task[T]], Initialize[Task[V]]] =
-    new Value[Seq[Task[T]], Initialize[Task[V]]] {
-      def appendValue(a: Seq[Task[T]], b: Initialize[Task[V]]): Seq[Task[T]] = ???
-    }
+    (_, _) => ???
```
Comment (eed3si9n) [inline main-settings/src/main/scala/sbt/Append.scala]: Same ask as before to comment here.

**Review by eed3si9n — CHANGES_REQUESTED** [review]

**Review by eed3si9n — APPROVED** [review]
Thanks

File: main-settings/src/test/scala/sbt/AppendSpec.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  Global / onLoad += { (s: State) =>
+    doSideEffect()
+    s
+  }
+
+  Global / onLoad += doSideEffect _
+  Global / onLoad += (() => doSideEffect())
+  Global / onLoad += (() => println("foo"))
```
Comment (2m) [inline main-settings/src/test/scala/sbt/AppendSpec.scala:30]: Neat!

Comment (dwijnand) [reply main-settings/src/test/scala/sbt/AppendSpec.scala:30]: Yeah, with sbt 1.3.0 we'll be able to do

```sbt
Global / onLoad += { s => dynverAssertTagVersion.value; s }
```

**Pull request merged**